Libraries/Dijkstra/Dijkstra.h

```diff
@@ -0,0 +1,262 @@
+#include "Map.h"
+
+struct coord {
+  uint8_t x, y;
+};
+
+struct dijkstra_tile {
+  uint8_t weight;
+  coord prev, current;
+  bool checked;
+};
+
+TVector<char> dijkstra(Map tiles_map) {
+  TVector<coord*> open_vector, non_visited_vector;
+  TVector<char> path;
+  dijkstra_tile matrix[tiles_map.getRows()][tiles_map.getColumns()];
+  coord *current_coord;
+  uint8_t current_row = tiles_map.getCurrRow();
+  uint8_t current_column = tiles_map.getCurrColumn();
+
+  //Setting all the elements of the matrix with the initial values
+  for (int i = 0 ; i < tiles_map.getRows() ; ++i) {
+    for (int j = 0 ; j < tiles_map.getColumns() ; ++j) {
+      matrix[i][j].current.x = i;
+      matrix[i][j].current.y = j;
+      matrix[i][j].prev.x = i;
+      matrix[i][j].prev.y = j;
+      matrix[i][j].weight = 255;
+      matrix[i][j].checked = false;
+      
+      if (tiles_map.getTile(i, j).getAccessible() == true && tiles_map.getTile(i, j).getVisited() == false && tiles_map.getTile(i, j).getBlack() == false) {
+        current_coord = &matrix[i][j].current;
+        non_visited_vector.pushTop(current_coord);
```

Comment:
Can you directly push the pointer? instead of assigning it to current_coord.

Reply:
I tried, but couldn't do that.